From 32ab9830ce577d87c4739acd15946754e3db0a51 Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@intel.com>
Date: Tue, 17 Oct 2017 15:02:50 +0200
Subject: [PATCH] connman: Update to 1.35

With binutils 2.27 on at least MIPS, connmand will crash on startup.  This
appears to be due to the symbol visibilty scripts hiding symbols that stdio
looks up at runtime, resulting in it segfaulting.

This certainly appears to be a bug in binutils 2.27 although the problem has
been known about for some time:

https://sourceware.org/bugzilla/show_bug.cgi?id=17908

As the version scripts are only used to hide symbols from plugins we can safely
remove the scripts to work around the problem until binutils is fixed.

Upstream-Status: Inappropriate
Signed-off-by: Ross Burton <ross.burton@intel.com>

---
 Makefile.am | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index e67a7a5..e9b868b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -129,8 +129,7 @@ src_connmand_LDADD = gdbus/libgdbus-internal.la $(builtin_libadd) \
 			@GLIB_LIBS@ @DBUS_LIBS@ @GNUTLS_LIBS@ \
 			-lresolv -ldl -lrt
 
-src_connmand_LDFLAGS = -Wl,--export-dynamic \
-				-Wl,--version-script=$(srcdir)/src/connman.ver
+src_connmand_LDFLAGS = -Wl,--export-dynamic
 
 src_connmand_wait_online_SOURCES = src/connmand-wait-online.c
 
@@ -173,8 +172,7 @@ vpn_connman_vpnd_LDADD = gdbus/libgdbus-internal.la $(builtin_vpn_libadd) \
 				@GLIB_LIBS@ @DBUS_LIBS@ @GNUTLS_LIBS@ \
 				-lresolv -ldl
 
-vpn_connman_vpnd_LDFLAGS = -Wl,--export-dynamic \
-				-Wl,--version-script=$(srcdir)/vpn/vpn.ver
+vpn_connman_vpnd_LDFLAGS = -Wl,--export-dynamic
 endif
 
 BUILT_SOURCES = $(local_headers) src/builtin.h $(service_files) \
